tests/CMakeLists.txt

```diff
@@ -11,6 +11,20 @@ if (NOT AUTOPAS_BUILD_TESTS)
     return()
 endif ()
 
+option(AUTOPAS_ENABLE_COVERAGE "Enable code coverage, requires GCC and CMAKE_BUILD_TYPE=Debug." OFF)
+message(STATUS "Enable code coverage: " ${AUTOPAS_ENABLE_COVERAGE})
+
+if (AUTOPAS_ENABLE_COVERAGE)
+  if (NOT AUTOPAS_BUILD_TESTS OR NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+    message(WARNING "Disabling code coverage.\n CMAKE_BUILD_TYPE=Debug and AUTOPAS_BUILD_TESTS=ON required.")
+    set(AUTOPAS_ENABLE_COVERAGE OFF CACHE BOOL "Enable code coverage, requires GCC and CMAKE_BUILD_TYPE=Debug." FORCE)
```

Comment:
I'd store the doc string in a variable for consistency

Reply:
done in [889f266](https://github.com/AutoPas/AutoPas/pull/888/commits/889f26618f64e9d66805dd320b2ef8d07f14578a)